src/http.h

```diff
@@ -63,7 +63,7 @@ class HttpStateData : public Client
     // Checks whether the response is cacheable/shareable.
     ReuseDecision::Answers reusableReply(ReuseDecision &decision);
 
-    CachePeer *_peer = nullptr;       /* CachePeer request made to */
+    CachePeer *peer() const { return fwd->serverConnection() ? fwd->serverConnection()->getPeer() : nullptr; }
```

Comment:
We should document new methods.

Let's use our serverConnection to reduce the number of repeated dereferences and, perhaps more importantly, avoid suspicions related to FwdState use of "its" serverConn, especially after we call `fwd->unregister()` and clear our serverConnection. This change also brings this code in sync with PR description.

```suggestion
    /// known cache_peer we are sending the request to (or nil)
    CachePeer *peer() const { return serverConnection ? serverConnection->getPeer() : nullptr; }

```

The above suggestion also separated the new method from data members.